### PR TITLE
CONSOLE-2892: Allow dynamic plugins to proxy to services on the cluster

### PR DESCRIPTION
**Pull request by jhadvig**
Console changes for story https://issues.redhat.com/browse/CONSOLE-2892

Console-operator will now check for `spec.proxy.services` of the enabled ConsolePlugins and manifest those services to the `console-config.yaml` ConfigMap, which will trigger redeploy the console pod to pick up the updates

eg:
```yaml
apiVersion: console.openshift.io/v1alpha1
kind: ConsolePlugin
metadata:
  name: console-demo-plugin
spec:
  displayName: 'OpenShift Console Demo Plugin'
  service:
    name: console-demo-plugin
    namespace: console-demo-plugin
    port: 9001
    basePath: '/'
  proxy:
    services:
    - name: thanos-querier
      namespace: openshift-monitoring
      port: 9091
      authorize: true
```

in the console browser window following query should be able to proxy the request to thanos and return 200:
```js
await fetch('/api/proxy/namespace/openshift-monitoring/service/thanos-querier:9091/api/v1/rules')
```

/assign @spadgett @TheRealJon @florkbr

### CODE REVIEW

File: pkg/console/subresource/configmap/configmap.go
Comment (spadgett) [reply pkg/console/subresource/configmap/configmap.go:131]: We don't want to pass the user's token around unless the service really needs it. Even if we trust the service, it's an exposure. The token could get accidentally logged, etc. It's much better to only add the `Authorization` header when we know we have to.

**Comment by jhadvig** [comment]
PR is ready for final review.

QE Approver:
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

Console Approver:
/assign @spadgett

File: pkg/console/subresource/configmap/configmap.go
Comment (spadgett) [inline pkg/console/subresource/configmap/configmap.go:143]: This is returning a URL, not a hostname.

**Review by spadgett — COMMENTED** [review]
Looks good to me, just one nit on a function name

**Comment by yapei** [comment]
verified the changes locally and works
/label qe-approved

**Comment by jhadvig** [comment]
@spadgett comments addressed in a new commit. PTAL

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
We need to get https://github.com/openshift/client-go/pull/199 in and update the go deps. accordingly.
/hold

**Comment by jhadvig** [comment]
GO deps have been updated. @spadgett please could you retag the PR. Thank you

/hold cancel

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**
